source/val/validation_state.cpp

```diff
@@ -394,5 +394,34 @@ void ValidationState_t::RegisterInstruction(
   if (id) {
     all_definitions_.insert(make_pair(id, &ordered_instructions_.back()));
   }
+
+  // If the instruction is using an OpTypeSampledImage as an operand, it should
+  // be recorded. The validator will ensure that all usages of an
+  // OpTypeSampledImage and its definition are in the same basic block.
+  for (auto i = 0; i < inst.num_operands; ++i) {
+    auto operand = inst.words[inst.operands[i].offset];
+    auto operand_inst = FindDef(operand);
```

Comment:
This must be guarded by a check that the operand type is SPV_OPERAND_TYPE_ID.

The proposed code can lead to false positives when the operand is, for example, a literal integer whose value happens to match the result id of an an OpSampledImage instruction.